src/core/entity.rs

```diff
@@ -0,0 +1,196 @@
+//! # PubNub entity module
+//!
+//! This module contains the [`PubNubEntity`] trait, which is used to implement
+//! a PubNub entity that can be used as a first-class citizen to access the
+//! [`PubNub API`].
+//!
+//! [`PubNub API`]: https://www.pubnub.com/docs
+
+use crate::{
+    lib::{
+        alloc::string::String,
+        core::{
+            cmp::PartialEq,
+            fmt::{Debug, Formatter, Result},
+        },
+    },
+    Channel, ChannelGroup, ChannelMetadata, UuidMetadata,
+};
+
+#[cfg(all(feature = "subscribe", feature = "std"))]
+use crate::{
+    core::{Deserializer, Transport},
+    lib::alloc::{sync::Arc, vec::Vec},
+    subscribe::{Subscribable, SubscribableType, Subscriber, Subscription, SubscriptionOptions},
+};
+
+pub(crate) trait PubNubEntity2 {
```

Comment:
`PubNubEntity2` ?

Reply:
This part is legacy. Need to tear it out.

Reply:
Removed